### PR TITLE
OEL-1421: Fix cosmetic changes on project

### PR DESCRIPTION
**Pull request by Maxfire**
Cosmetic changes and tests provided.

### CODE REVIEW

File: config/install/field.field.node.oe_project.oe_cx_impacts.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ dependencies:
 third_party_settings:
   allowed_formats:
     allowed_formats:
-      - simple_rich_text
+      - rich_text
```
Comment (abel-santos-corral) [inline config/install/field.field.node.oe_project.oe_cx_impacts.yml:13]: Please, check if tests for this change can be done. See the model of SearchTextTest.php file at oe_whitelabel_paragraphs submodile at OE Whitelabel. E.g.     $text = $crawler->filter('p#paragraph-2 strong');

**Review by abel-santos-corral — CHANGES_REQUESTED** [review]

**Review by abel-santos-corral — DISMISSED** [review]

File: tests/src/ExistingSite/ProjectTest.php [hunk 2]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
     $assert_session->pageTextContains('Belgium');
     $assert_session->pageTextContains('Contribution to the budget');
     $assert_session->pageTextContains('€22,90');
+    $assert_session->pageTextContains('Project details');
+
+    $this->drupalLogin($this->createUser([], '', TRUE));
+    // Create a node, without project details.
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: rather where a project is first created and we fill in the project details (starting line 73), **don't**, assert what you wanted from the project detaild group and then, edit the node, add the values, and assert that the values and group are now there

File: tests/src/ExistingSite/ProjectTest.php [hunk 3]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
...
+
+    // Assert rich text.
+    $assert_session->hiddenFieldValueEquals('oe_summary[0][format]', 'rich_text');
+    $assert_session->hiddenFieldValueEquals('oe_cx_objective[0][format]', 'rich_text');
+    $assert_session->hiddenFieldValueEquals('oe_cx_impacts[0][format]', 'rich_text');
+    $assert_session->hiddenFieldValueEquals('oe_cx_achievements_and_milestone[0][format]', 'rich_text');
+
+    // Participants.
+    $page->pressButton('Add new participant');
+    $page->fillField('oe_project_participants[form][0][name][0][value]', 'Developer participant name');
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: add contribution

File: tests/src/ExistingSite/ProjectTest.php [hunk 4]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
...
+    $page->pressButton('Create participant');
+
+    // Lead contributors.
+    $page->pressButton('edit-oe-cx-lead-contributors-actions-ief-add');
+    $page->fillField('oe_cx_lead_contributors[form][0][name][0][value]', 'Lead contributors name');
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: add contribution

File: tests/src/ExistingSite/ProjectTest.php [hunk 5]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
...
+    $page->pressButton('Create organisation');
+    $page->pressButton('Save');
+
+    $this->drupalGet('project/project-page-test-w-details');
+    // Check Project details is not present when empty.
+    $assert_session->pageTextNotContains('Project details');
+    // Check the correct order.
+    $correct_order = [
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: either check the order of everything or dont check order

File: tests/src/ExistingSite/ProjectTest.php [hunk 6]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
...
+    // 1st node contains the period.
+    $assert_session->elementExists('css', '.card-body:nth-child(1) .me-4-5');
+    // 2nd node doesn't contain any period, therefore no empty span.
+    $assert_session->elementNotExists('css', '.card-body:nth-child(2) .me-4-5');
+  }
+
+  /**
+   * Create a project listing page.
+   */
+  protected function createView(): void {
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: remove this

File: tests/src/ExistingSite/ProjectTest.php [hunk 7]
```diff
@@ -38,6 +39,7 @@ public function testCreateProject() {
     $this->markEntityTypeForCleanup('node');
     $this->markEntityTypeForCleanup('media');
     $this->markEntityTypeForCleanup('file');
+    $this->markEntityTypeForCleanup('view');
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: remove this

File: tests/src/ExistingSite/ProjectTest.php [hunk 8]
```diff
@@ -165,6 +167,110 @@ public function testCreateProject() {
...
+      2 => 'Participants',
+    ];
+    foreach ($correct_order as $key => $value) {
+      $assert_session->elementContains('xpath', "(//ul[contains(@class, 'nav-pills')]//li[contains(@class, 'nav-item')])[" . $key . "]", $value);
+    }
+
+    // Create a project listing view page.
+    $this->createView();
+    $this->container->get('router.builder')->rebuildIfNeeded();
+    $this->drupalGet('/project-listing-page');
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: just use /node

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/ExistingSite/ProjectTest.php (hunk 4 above)
Comment (Maxfire) [reply tests/src/ExistingSite/ProjectTest.php]: no need cause I'm going to edit the previous node which already have contribution.

File: tests/src/ExistingSite/ProjectTest.php (hunk 3 above)
Comment (Maxfire) [reply tests/src/ExistingSite/ProjectTest.php]: no need cause I'm going to edit the previous node which already have contribution.

File: tests/src/ExistingSite/ProjectTest.php [hunk 9]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
...
+    // Check the correct order.
+    $correct_order = [
+      1 => 'Summary',
+      2 => 'Objective',
+      3 => 'Impacts',
+      4 => 'Lead contributors',
+      5 => 'Participants',
+    ];
+    foreach ($correct_order as $key => $value) {
+      $assert_session->elementContains('xpath', "(//ul[contains(@class, 'nav-pills')]//li[contains(@class, 'nav-item')])[" . $key . "]", $value);
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: can't we use li.nav-item:nth-child as selector, then getText and compare ? would be clearer

File: tests/src/ExistingSite/ProjectTest.php [hunk 10]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
...
+    $page->fillField('oe_project_funding_programme[0][target_id]', '');
+    $page->fillField('Reference', '');
+    $page->pressButton('Remove');
+    $page->pressButton('Remove');
+    $page->pressButton('Save');
+
+    $this->drupalGet('project/project-page-test');
+    // Check Project details is not present when empty.
+    $assert_session->pageTextNotContains('Project details');
+    // Check the correct order.
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: Rather say your asserting the in-page nav, and if you do this here do this also above, when you have all items present, so we can see that it changed

File: tests/src/ExistingSite/ProjectTest.php [hunk 11]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
...
+      3 => 'Impacts',
+      4 => 'Lead contributors',
+      5 => 'Participants',
+    ];
+    foreach ($correct_order as $key => $value) {
+      $assert_session->elementContains('xpath', "(//ul[contains(@class, 'nav-pills')]//li[contains(@class, 'nav-item')])[" . $key . "]", $value);
+    }
+
+    $this->drupalGet('/node');
+    dump($page->getHtml());
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: left over, remove

File: tests/src/ExistingSite/ProjectTest.php [hunk 12]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
...
+    // Node doesn't contain any period, therefore no empty span.
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php]: I dont understand what we are asserting here

File: tests/src/ExistingSite/ProjectTest.php [hunk 13]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
...
+    $assert_session->elementNotExists('css', '.card-body:nth-child(1) .text-muted');
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php:236]: we check that no meta is present, can we check somewhere also a case where it is

File: tests/src/ExistingSite/ProjectTest.php [hunk 14]
```diff
@@ -165,6 +165,51 @@ public function testCreateProject() {
     $assert_session->pageTextContains('Belgium');
     $assert_session->pageTextContains('Contribution to the budget');
     $assert_session->pageTextContains('€22,90');
+    $assert_session->pageTextContains('Project details');
+
```
Comment (drishu) [inline tests/src/ExistingSite/ProjectTest.php:188]: I see a lot of "Improve the field locator." could you do me a favour, remove them all and add just one todo in the function comment explaining that we want to use labels not machine names in the future

**Review by drishu — CHANGES_REQUESTED** [review]

File: tests/src/ExistingSite/ProjectTest.php (hunk 13 above)
Comment (Maxfire) [reply tests/src/ExistingSite/ProjectTest.php:236]: we are, just a bit up.

**Review by drishu — DISMISSED** [review]

**Review by abel-santos-corral — DISMISSED** [review]

**Review by abel-santos-corral — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**